### PR TITLE
Add coverage to C++ tests

### LINKED ISSUE #223: Add code coverage
We should see what the code coverage is like.

### PR DESCRIPTION
**Pull request by mgeplf**
* Add cmake option MORPHIO_TESTS
* Have coverage created for Debug build tests
* add github action for uploading coverage
* fixes #223

### CODE REVIEW

File: ci/coverage_test.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+rm -rf "$BUILD_DIR"
+mkdir -p "$BUILD_DIR"
+cd "$BUILD_DIR"
+cmake                           \
+    -DCMAKE_BUILD_TYPE=Debug    \
+    -DHIGHFIVE_EXAMPLES=OFF     \
+    -DHIGHFIVE_UNIT_TESTS=OFF   \
+    -G"Unix Makefiles"          \
+    ../..
```
Comment (tomdele) [inline ci/coverage_test.sh:17]: isn't it one step too low ?

Comment (tomdele) [reply ci/coverage_test.sh:17]: my bad sorry missed the `build/build-coverage`

**Review by tomdele — APPROVED** [review]
I am good with the changes. 
What s the current C++ coverage ?

**Comment by mgeplf** [comment]
Available here:
https://coveralls.io/github/BlueBrain/MorphIO

File: .github/workflows/coverage_test.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: coverage-test
+on: [pull_request, push]
+
+jobs:
+    build:
+        name: coverage-test
+        runs-on: ubuntu-18.04
```
Comment (matz-e) [inline .github/workflows/coverage_test.yaml:7]: No 20.04?

File: CMake/CodeCoverage.cmake [hunk 3]
```diff
@@ -0,0 +1,303 @@
+# Copyright (c) 2012 - 2017, Lars Bilke
```
Comment (matz-e) [inline CMake/CodeCoverage.cmake:1]: Seems like a lot of copy-pasta. I suppose there's no better place to include this from.

File: ci/coverage_test.sh (hunk 1 above)
Comment (matz-e) [reply ci/coverage_test.sh:17]: `pushd` / `popd` would be an alternative to random amounts of `..`.

**Review by matz-e — APPROVED** [review]
My 0.02 BTC

File: CMake/CodeCoverage.cmake (hunk 3 above)
Comment (mgeplf) [reply CMake/CodeCoverage.cmake:1]: yeah, it's versioned in; small amount of code like this I find it better

File: .github/workflows/coverage_test.yaml (hunk 2 above)
Comment (mgeplf) [reply .github/workflows/coverage_test.yaml:7]: not yet

**Pull request merged**